osf/models/sanctions.py

```diff
@@ -308,16 +312,20 @@ def _email_template_context(self, user, node, is_authorizer=False):
         return {}
 
     def _notify_authorizer(self, authorizer, node):
+        if self.should_suppress_emails:
+            return
         context = self._email_template_context(authorizer,
-                                               node,
-                                               is_authorizer=True)
+                                            node,
+                                            is_authorizer=True)
         if self.AUTHORIZER_NOTIFY_EMAIL_TEMPLATE:
             self._send_approval_request_email(
                 authorizer, self.AUTHORIZER_NOTIFY_EMAIL_TEMPLATE, context)
         else:
             raise NotImplementedError
 
     def _notify_non_authorizer(self, user, node):
+        if self.should_suppress_emails:
```

Comment:
Can we just make this function conditional in the `ask` method? Seems like bad practice enter this function only to exit immediately.